.github/workflows/integration-test.yml

```diff
@@ -0,0 +1,49 @@
+name: Integration tests
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container:
+        # TODO: what's better place for this test base docker image?
+        image: docker.io/ochaloup/solana-test-validator:1.14.18
```

Comment:
what's the actual size of this image? is it trimmed? If it is small I think we can add it in our ECR or maybe there's a public one already?

Reply:
👍 Please do not use image in your personal space on github - we got our container repository. The image should ideally be included in the pipeline in our IaaC.

Reply:
@butonium The image is pretty big (currently ~2.5GB). I was not trying to optimize for the size, I wanted just get the solana tools and avm anchor being installed.

If you give me some ideas on steps to check to trimming the image I can try. While I'm not sure if it's really useful as the github actions seems not having issue to download and cache this kind of image either way. I think there is not a plan to use it elsewhere.

Maybe there could be a way to slim as it's done in like `ops-infra` (like https://github.com/marinade-finance/ops-infra/blob/master/jenkins/dockerfiles/rust-anchor-0.26.0-pnpm.Dockerfile) and experiment in copying only binaries of the anchor and solana which could be quite smaller. But I don't know to spend investigation time on that as I don't know it on top of my head.

On public part. I haven't found what can be useful for me. There is the project serum 
https://hub.docker.com/layers/projectserum/build/v0.27.0/images/sha256-0e11aced57c448c7da9bf0a563a146398ac8b88f357b8fc2e1be314b42320686?context=explore
but it did not work properly (and still the 0.28.0 is not released).

By interest, when it's pushed in Marinade ECR then it's publicly available or the github actions has got permission with some app key?

Reply:
We can make it publicly available. Rule of thumb is to just put in the image what you need. You will easily get to 100 - 200 MB.

Reply:
@janlegner I see, just I was a bit lazy to work through that. But let's see, I will check what I can do about it.

Reply:
No worries

Reply:
@janlegner With a pretty while of playing I slimmed the image size to ~280MB. It can be hardly significantly less as the solana binaries (while limited only to needed part) takes around 150 MB.
On top, squashing may bring an image smaller of 30MB but I would think it's beneficial to use it.
Would the size around 300MB be ok?

Reply:
@ochaloup perfect, thanks